opentelemetry-user-events-trace/src/trace/exporter.rs

```diff
@@ -0,0 +1,167 @@
+            ))
+        }
+    }
+}
+
+impl UserEventsSpanExporter {
+    /// Create a new instance of the exporter
+    pub(crate) fn new(provider_name: &str) -> Result<Self, String> {
+        if provider_name.len() >= 234 {
+            return Err("Provider name must be less than 234 characters.".to_string());
+        }
+        if !provider_name
+            .chars()
+            .all(|c| c.is_ascii_alphanumeric() || c == '_')
+        {
+            return Err(
+                "Provider name must contain only ASCII letters, digits, and '_'.".to_string(),
+            );
+        }
+
+        let mut eventheader_provider = Provider::new(provider_name, &Provider::new_options());
+        let keyword = 1;
+        let event_set = eventheader_provider.register_set(Level::Informational, keyword);
+        let name = eventheader_provider.name().to_string();
+
+        Ok(UserEventsSpanExporter {
+            provider: Mutex::new(eventheader_provider),
+            name,
+            event_set,
+        })
+    }
+
+    fn add_attribute_to_event(&self, eb: &mut EventBuilder, kv: &KeyValue) {
+        let field_name = kv.key.as_str();
+        match &kv.value {
+            Value::Bool(b) => {
+                eb.add_value(field_name, *b, FieldFormat::Boolean, 0);
+            }
+            Value::I64(i) => {
+                eb.add_value(field_name, *i, FieldFormat::SignedInt, 0);
+            }
+            Value::F64(f) => {
+                eb.add_value(field_name, *f, FieldFormat::Float, 0);
+            }
+            Value::String(s) => {
+                eb.add_str(field_name, s.as_str(), FieldFormat::Default, 0);
+            }
+            _ => (),
+        }
+    }
+
+    pub(crate) fn export_span(&self, span: &SpanData) -> OTelSdkResult {
+        if self.event_set.enabled() {
+            let mut eb = EventBuilder::new();
+            eb.reset("Span", 0);
+            eb.opcode(Opcode::Info);
+            eb.add_value("__csver__", 1024, FieldFormat::UnsignedInt, 0);
+
+            eb.add_struct("PartA", 2, 0); // time, ext_dt
+            let datetime: DateTime<Utc> = span.end_time.into();
+            eb.add_str("time", datetime.to_rfc3339(), FieldFormat::Default, 0);
+            eb.add_struct("ext_dt", 2, 0);
+            eb.add_str(
+                "traceId",
+                span.span_context.trace_id().to_string(),
+                FieldFormat::Default,
+                0,
+            );
+            eb.add_str(
+                "spanId",
+                span.span_context.span_id().to_string(),
+                FieldFormat::Default,
+                0,
+            );
+
+            eb.add_struct("PartB", 5, 0);
+            eb.add_str("_typeName", "Span", FieldFormat::Default, 0);
+            eb.add_str("name", span.name.as_ref(), FieldFormat::Default, 0);
+            eb.add_str(
+                "parentId",
+                span.parent_span_id.to_string(),
```

Comment:
this should be empty/not-emitted when this is a root span?

Reply:
Have made it empty for root span for now.